src/context.rs

```diff
@@ -319,14 +319,16 @@ impl<A: Actor> Context<A> {
         self.tick(self.recv_once().await, act).await;
     }
 
-    /// Handle any incoming messages for the actor while running a given future.
+    /// Joins on a future by handling all incoming messages whilst polling it. The future will
+    /// always be polled to completion, even if the actor is stopped. If the actor is stopped,
+    /// handling of messages will cease, and only the future will be polled.
```

Comment:
In other words, this behaves similar to `futures::join` which waits on both futures to complete, right?

Reply:
Yes - the one difference is that the incoming stream of messages does not usually complete. So, it's like join, but only the user future *must* complete for it to return

Reply:
One could argue that an actor stopping is not necessarily a failure case so it can definitely happen.

Reply:
It can happen, sure. What I mean, though, is that if this were semantically equivalent to join(handle_messages, future), then it would not return until the actor stopped. This function, by comparison, will return if the actor doesn't stop if future does.

Reply:
Ah right! Thanks for the clarification.

I guess `join` is still a pretty good name for this although when comparing it to `futures::join` it is not quite accurate.

If we ever get custom event loops, I think this could be removed in favour of the user polling for a new message themselves which then does away with the naming problem :)

Reply:
True. It might still be worth keeping this around in some form for convenience - maybe in an extension crate, though. When custom event loops happen, hopefully the function will only be a few lines, too!

Reply:
The user polling on their own could have potential for mis-ordered messages; I will write up how and why in #85 when it's ready for review